### ISSUE TITLE
IPython on OSX -- __main__.py: error: argument --framerate/-f: invalid int value

### ISSUE THREAD

**Issue by p-i-** [glumpy/glumpy#88]
In an IPython cell (OSX) I'm attempting to run the following:

```
import glumpy

z = ... some 2D float array ...
window = glumpy.Window(512, 512)
im = glumpy.Image(z.astype(np.float32), cmap=glumpy.colormap.Grey)

@window.event
def on_draw():
    im.blit(0, 0, window.width, window.height)
window.mainloop()
```

This is generating:
```
usage: __main__.py [-h]
                   [--backend {glfw,sdl2,qt5,pyside,pyglet,sdl,freeglut,osxglut}]
                   [--record] [--interactive] [--framerate FRAMERATE]
                   [--display-fps] [--debug] [--size SIZE]
                   [--position POSITION] [--single-buffer] [--stereo]
                   [--vsync VSYNC] [--srgb] [--depth-size DEPTH_SIZE]
                   [--stencil-size STENCIL_SIZE] [--gl-api {GL,ES}]
                   [--gl-profile {none,core,compatibility}]
                   [--gl-version GL_VERSION]
__main__.py: error: argument --framerate/-f: invalid int value: '/Users/pi/Library/Jupyter/runtime/kernel-166ad1a3-d108-4ccc-97cc-56d7e264edbe.json'
```

It's difficult to see what is going on here. It would appear that glumpy is attempting to start itself up but passing an invalid value (i.e. that path string) for the `framerate` parameter.

Might this be a bug?

**Comment by rougier** [comment]
The example you mention may come from an earlier version of glumpy. You light need to use latest glumpy and look at https://github.com/rougier/glumpy/tree/master/examples.

**Comment by rougier** [comment]
And glumpy is not compatible with the jupyter notebook. But you can use the `--interactive` flag if you want to interact with your visualization from the console.

**Comment by lumpen** [comment]
@rougier the lastest version glumpy is also not compatible with the jupyter notebook?

**Comment by rougier** [comment]
Yes, unfortunately still not compatible. It would require not too much work I think but I'm not competent in javascript.

**Comment by EricCousineau-TRI** [comment]
While I understand that figuring out the Javascript shennanigans would be difficult for remote use, might there still be a way to at least monkey-patch the `argparse`-logic in `glumpy` to not use `argparse`, so that it doesn't try to read `sys.argv`?
In this way, `glumpy` could still be used in Jupyter in local kernels and shell out to nominal OpenGL calls.

(I'll briefly try this out and see what happens...)

**Comment by EricCousineau-TRI** [comment]
This is what worked for me:
```py
import glumpy.app.parser as m

def get_options_mock():
    mock_argv = []
    options, unknown = m.get_default().parse_known_args(mock_argv)
    return options

m.get_options = get_options_mock
```

That being said, this just addresses the case of using off-screen rendering. Of course, this won't work for interactive use cases. (Sorry!)

**Comment by rougier** [comment]
Sorry for late answer. Could we generalize your solution for any glumpy script ? I developed the automatic argument parsing for ease of use but I realize it might bring troubles in some cases.  Maybe we can add a specific import (e.g. import `glumpy.no_argparse`) to disable args parsing.
